packages/core/src/App.ts

```diff
@@ -171,6 +171,10 @@ export class App extends Module {
 		try {
 			await this.onInit?.(this);
 			await mapSeries(this.modules, async module => {
+				if (module.getStatus() === 'destroyed' || module.getStatus() === 'error') {
+					module.setStatus('registering');
```

Comment:
it's better to run the onRegister hook for all the modules first, and then afterwards to run the onInit for all the modules.